### PR TITLE
cut 2.13.0-rc.0 pre-release

### PR DESCRIPTION
**Pull request by krasi-georgiev**
@brian-brazil @juliusv @beorn7 @gouthamve @bwplotka @codesome @SuperQ @csmarchbanks @brancz @simonpasquier

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,22 @@
+## 2.13.0-rc.0 / 2019-09-25
+
+* [CHANGE] Metrics: Change prometheus_sd_configs_failed_total to Gauge #5254
+* [CHANGE] Service discovery: add new node address types for kubernetes. #5902
+* [CHANGE] Metrics format: update parsing to disallow new Go 1.13 float formats. #5984
```
Comment (simonpasquier) [inline CHANGELOG.md]: not sure this needs to be in the changelog since it doesn't change anything.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+* [ENHANCEMENT] Remote read: use remote read streaming to reduce memory usage. #5703
+* [ENHANCEMENT] Docs: fix spelling mistakes and other small changes . #5952, #6051
```
Comment (simonpasquier) [inline CHANGELOG.md]: i don't think we list those changes usually.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,22 @@
...
+* [ENHANCEMENT] GUI: show warnings if query have returned some warnings. #5964
+* [ENHANCEMENT] Reduce memory usage: cache labels.Labels to Identify Series in Remote Write. #5849
+* [ENHANCEMENT] Metrics: added metrics for remote write max/min/desired shards to queue manager. #5787
+* [ENHANCEMENT] Promtool: show the warnings during label query. #5924
+* [ENHANCEMENT] Promtool: improve error messages when parsing bad rules. #5965
+* [ENHANCEMENT] Promtool: more promlint rules. #5515
+* [ENHANCEMENT] Promtool: fix recording inconsistency due to duplicate labels. #6026
+* [BUGFIX] GUI: fixes service-discovery view when accessed from unhealthy targets. #5915
+* [BUGFIX] Service discovery example: when ip out of order, reflect.deepEqual can not correctly identify whether there is a change. #5856
```
Comment (simonpasquier) [inline CHANGELOG.md]: this shouldn't be listed as the custom SD code isn't part of the release.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,22 @@
+## 2.13.0-rc.0 / 2019-09-25
+
+* [CHANGE] Metrics: Change prometheus_sd_configs_failed_total to Gauge #5254
+* [CHANGE] Service discovery: add new node address types for kubernetes. #5902
+* [CHANGE] Metrics format: update parsing to disallow new Go 1.13 float formats. #5984
+* [ENHANCEMENT] Remote read: use remote read streaming to reduce memory usage. #5703
+* [ENHANCEMENT] Docs: fix spelling mistakes and other small changes . #5952, #6051
+* [ENHANCEMENT] GUI: show warnings if query have returned some warnings. #5964
+* [ENHANCEMENT] Reduce memory usage: cache labels.Labels to Identify Series in Remote Write. #5849
```
Comment (simonpasquier) [inline CHANGELOG.md]: (suggestion) `Remove write: reduce memory usage by caching labels to identify series.`

File: CHANGELOG.md (hunk 2 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: yea probably right, bigger docs changes would be good to list, but these small ones are not helpful.

File: CHANGELOG.md (hunk 4 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: good suggestion, thanks

File: CHANGELOG.md (hunk 3 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: it would be in the Source code zip file no?

File: CHANGELOG.md (hunk 1 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: doesn't this mean that the exporters should change how they export floats?

File: CHANGELOG.md (hunk 3 above)
Comment (codesome) [reply CHANGELOG.md]: The release notes usually contain only user-facing changes.

File: CHANGELOG.md (hunk 1 above)
Comment (brian-brazil) [reply CHANGELOG.md]: No, there's no changes here. This doesn't need to be mentioned.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,22 @@
+## 2.13.0-rc.0 / 2019-09-25
+
+* [CHANGE] Metrics: Change prometheus_sd_configs_failed_total to Gauge #5254
+* [CHANGE] Service discovery: add new node address types for kubernetes. #5902
```
Comment (brian-brazil) [inline CHANGELOG.md]: Is this really a CHANGE?

File: CHANGELOG.md (hunk 2 above)
Comment (brian-brazil) [reply CHANGELOG.md]: Even big doc changes don't get listed, as they don't affect the binary

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,22 @@
...
+* [CHANGE] Metrics format: update parsing to disallow new Go 1.13 float formats. #5984
+* [ENHANCEMENT] Remote read: use remote read streaming to reduce memory usage. #5703
+* [ENHANCEMENT] Docs: fix spelling mistakes and other small changes . #5952, #6051
+* [ENHANCEMENT] GUI: show warnings if query have returned some warnings. #5964
```
Comment (brian-brazil) [inline CHANGELOG.md]: It's a UI, we don't have a GUI

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,22 @@
...
+* [ENHANCEMENT] Reduce memory usage: cache labels.Labels to Identify Series in Remote Write. #5849
+* [ENHANCEMENT] Metrics: added metrics for remote write max/min/desired shards to queue manager. #5787
+* [ENHANCEMENT] Promtool: show the warnings during label query. #5924
+* [ENHANCEMENT] Promtool: improve error messages when parsing bad rules. #5965
+* [ENHANCEMENT] Promtool: more promlint rules. #5515
+* [ENHANCEMENT] Promtool: fix recording inconsistency due to duplicate labels. #6026
```
Comment (brian-brazil) [inline CHANGELOG.md]: This sounds like a BUGFIX

File: CHANGELOG.md (hunk 3 above)
Comment (brian-brazil) [reply CHANGELOG.md]: Yes, this doesn't get included. It's not part of the release.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,22 @@
+## 2.13.0-rc.0 / 2019-09-25
+
+* [CHANGE] Metrics: Change prometheus_sd_configs_failed_total to Gauge #5254
```
Comment (csmarchbanks) [inline CHANGELOG.md]: Consider mentioning that the name is now `prometheus_sd_failed_configs`.

File: CHANGELOG.md (hunk 5 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: ENHANCEMENT ?

File: CHANGELOG.md (hunk 4 above)
Comment (csmarchbanks) [reply CHANGELOG.md]: We were already caching the labels, this PR just changed the internal format. Perhaps:
`Remote write: reduce memory usage of the series cache.`

Comment (krasi-georgiev) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 3 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 2 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: removed

File: CHANGELOG.md (hunk 1 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: removed

File: CHANGELOG.md (hunk 8 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 7 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 6 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 5 above)
Comment (krasi-georgiev) [reply CHANGELOG.md]: @simonpasquier could you give some context, should this be included and is it a change or an enhancment?

Comment (krasi-georgiev) [reply CHANGELOG.md]: updated to enhancment

**Pull request merged**